xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml

```diff
@@ -0,0 +1,1144 @@
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'desc')
+  #else
+    #set ($sortOrder = 'asc')
+  #end
+#elseif ($confluenceSortOrder == 'descending')
+  #if ($reverseOrderProperties.contains($sortBy))
+    #set ($sortOrder = 'asc')
+  #else
+    #set ($sortOrder = 'desc')
+  #end
+#else
+{{warning}}
+Warning: sortOrder parameter must be one of 'ascending' or 'descending'
+{{/warning}}
+#end
+#set ($attachmentsFiltered = $collectiontool.sort($attachmentsFiltered, "$sortBy:$sortOrder"))
+## Display attchments
+{{html clean="false"}}
+#if ($attachmentsFiltered.size() &gt; 0)
+  #displayAttachments($attachmentsFiltered)
+  #deleteAttachmentModal()
+#else
+  &lt;p class="noitems"&gt;$services.localization.render('core.viewers.attachments.noAttachments')&lt;/p&gt;
+#end
+{{/html}}
+## Allow upload
+#set ($upload = "$!wikimacro.parameters.upload")
+#if ($upload == 'true' &amp;&amp; ($hasEdit || $hasAdmin) &amp;&amp; $xcontext.action == 'view')
+
+
+  {{html}}
+    &lt;form action="$doc.getURL("upload")" enctype="multipart/form-data" method="post" id="AddAttachment"&gt;
+    &lt;div&gt;
+    ## CSRF prevention
+    &lt;input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" /&gt;
+    ## $redirect is defined in attachmentslist.vm
+    &lt;input type="hidden" name="xredirect" value="$escapetool.xml($redirect)" /&gt;
+    &lt;fieldset id="attachform"&gt;
+      &lt;legend&gt;$services.localization.render('core.viewers.attachments.upload.title')&lt;/legend&gt;
+      ## Temporarily disabled, until we fix attachment name handling
+      ## &lt;div&gt;&lt;label id="xwikiuploadnamelabel" for="xwikiuploadname"&gt;$services.localization.render('core.viewers.attachments.upload.filename')&lt;/label&gt;&lt;/div&gt;
+      ## &lt;div&gt;&lt;input id="xwikiuploadname" type="hidden" name="filename" value="" size="40"/&gt;&lt;/div&gt;
+      &lt;div class="fileupload-field"&gt;
+        &lt;label class="hidden" for="xwikiuploadfile"&gt;$services.localization.render('core.viewers.attachments.upload.file')&lt;/label&gt;
+        &lt;input id="xwikiuploadfile" type="file" name="filepath" size="40" class="uploadFileInput noitems" data-max-file-size="$!escapetool.xml($xwiki.getSpacePreference('upload_maxsize'))" /&gt;
+      &lt;/div&gt;
+      &lt;div&gt;
+        &lt;span class="buttonwrapper"&gt;
+          &lt;input type="submit" value="$services.localization.render('core.viewers.attachments.upload.submit')" class="button btn btn-primary"/&gt;
+        &lt;/span&gt;
+        &lt;span class="buttonwrapper"&gt;
+          &lt;a class="cancel secondary button btn btn-primary" href="$doc.getURL()"&gt;$services.localization.render('core.viewers.attachments.upload.cancel')&lt;/a&gt;
+        &lt;/span&gt;
+      &lt;/div&gt;
+    &lt;/fieldset&gt;
+    &lt;/div&gt;
+    &lt;/form&gt;
+  {{/html}}
+#end
+{{/velocity}}
+
+{{html clean="false"}}
+  &lt;/div&gt; &lt;!-- .confluence-attachment-macro --&gt;
```

Comment:
I'd rather keep this in the previous HTML macro and also inside Velocity, thus using a Velocity comment instead of generating a useless XML comment (on the client side).